packages/pixelboy/apps/journal/checkin/checkin.coffee

```diff
@@ -47,3 +49,34 @@ class PAA.PixelBoy.Apps.Journal.CheckIn extends AM.Component
 
       else
         FlowRouter.go 'journal'
+
+  getExternalImgUrl: (event) ->
+    url = event.currentTarget.value
+    externalImgUrl = null
+    imgPreview = $('.image-preview')
+    placeholder = 'http://placehold.it/350x150?text=your+art'
+    errorIndicator = $('.preview-error')
+
+    if /twitter\.com/.test(url)
```

Comment:
Before or inside major branching points include a comment. It can be something like `# Handle Twitter post.` or `# Get images from a Tweet.`

I generally like to leave a lot of comments, even if they seem to say the same as the code. But when you look at the code there's at least some 'decoding' time, whereas comments are directly literal. Another way to look at it is if you imagine that you'd write all comments in advance before you started coding, you would lay out comments (pseudocode) as if it was a recipe with major points of the algorithms listed out. But it's a hard thing to judge and adapt your usual commenting style, so just look at other code and slowly it should go in that direction.